nUpdate/Updating/UpdateManager.cs

```diff
@@ -309,6 +312,11 @@ public bool SearchForUpdates()
             return true;
         }
 
+        void result_MissingRequirements(object sender, FailedEventArgs e)
```

Comment:
Missing access modifier and the name of the handler should be changed to "MissingRequirements".
Also the FailedEventArgs are not really the way of doing that here. I would create new EventArgs (MissingRequirementEventArgs) that have a property **public List<UpdateRequirement> MissingRequirements {get; set; }** and are used and handled by the new dialog you've created.